auth/src/main/java/org/aerogear/mobile/security/AbstractSecurityCheckExecutor.java

```diff
@@ -5,65 +5,74 @@
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
+import org.aerogear.mobile.core.executor.AppExecutors;
 import org.aerogear.mobile.core.metrics.MetricsService;
 
 import java.util.Collection;
 import java.util.HashSet;
 
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
+
 /**
+ *
  * Base class for security check executors.
+ *
+ * @param <T> the real executor type to be returned to chain the adding of checks to be executed
  */
 abstract class AbstractSecurityCheckExecutor<T extends AbstractSecurityCheckExecutor> {
 
-    /**
-     * Collection of checks to be executed.
-     */
     private final Collection<SecurityCheck> checks = new HashSet<>();
-
-    /**
-     * Context.
-     */
     private final Context context;
-
-    /**
-     * Metric singleThreadService to be used to publish metrics. Can be null.
-     */
     private final MetricsService metricsService;
 
     /**
-     * Constructor.
+     * Creates AbstractSecurityCheckExecutor object.
      *
-     * @param context the context
-     * @param checks checks to be executed
-     * @param metricService singleThreadService to be used to publish metrics. If null, no metrics get published.
+     * @param context {@link Context} to be used by the security checks
+     * @param checks  {@link Collection} of {@link SecurityCheck} to be executed
+     * @param metricService {@link MetricsService} is a {@link AppExecutors#singleThreadService()}, to be used to publish metrics. If null, no metrics get published
```

Comment:
Didn't notice this: I don't think metricService is a AppExecutors#singleThreadService().
MetricService si a {@link MetricsService}.